templates/cc/string.go

```diff
@@ -135,12 +144,39 @@ const strTpl = `
 		}
 		*/}}
 	{{ else if $r.GetHostname }}
-		{{ unimplemented }}
-		{{/* TODO(akonradi) implement hostname constraints
-		if err := m._validateHostname({{ accessor . }}); err != nil {
-			return {{ errCause . "err" "value must be a valid hostname" }}
+	{
+		// Take a copy so we can lowercase it.
+		std::string value = {{ accessor . }};
+		std::transform(value.begin(), value.end(), value.begin(), [](unsigned char c){ return std::tolower(c); });
```

Comment:
What's the use of making this lower-case? Why not just allow uppercase characters below? That way we can avoid an extra copy.

Reply:
To be honest this is copied from the golang implementation. I'm guessing the they lowercase it to save an extra: "||" branch. I'm guessing it'll just be cheaper to do that check though, rather than lowercase a potentially large string.